src/jupyter_contrib_nbextensions/nbextensions/execution_dependencies/execution_dependencies.js

```diff
@@ -0,0 +1,96 @@
+define([
+    'base/js/namespace',
+    'notebook/js/codecell'
+], function (
+    Jupyter,
+    codecell
+) {
+    "use strict";
+
+    var CodeCell = codecell.CodeCell;
+
+    return {
+        load_ipython_extension: function () {
+            console.log('[exec_deps] patching CodeCell.execute');
+            var orig_execute = codecell.CodeCell.prototype.execute;                                                 // get original cell execute function
+            CodeCell.prototype.execute = function (stop_on_error) {
+                var root_tags = this.metadata.tags || [];
+                if(root_tags != [] && root_tags.some(tag => /=>.*/.test(tag))) {                                    // if the root cell contains any dependencies, resolve dependency tree...
```

Comment:
I was not sure if this.metadata.tags would be [] or something like null/undefined. That is where my javascript knowledge ends. But I see that since I either have [...] or [], so it is defined and I can ask for its length.

Reply:
Exactly. `this.metadata.tags` may often be `undefined`, e.g. if no tags have been set. In fact, in a trickier world, it could be anything that'll fit into the notebook JSON, but in practice we can assume that it will be either an array of strings, or undefined (if it's not, something more significant has gone wrong with the notebook somewhere, and this extension breaking is likely to be the least of anyone's worries)